### PR TITLE
Add retroarch metal nightly

### PR DESCRIPTION
**Pull request by fpscan**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/r/retroarch-metal-nightly.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+cask "retroarch-metal-nightly" do
```
Comment (bevanjkay) [inline Casks/r/retroarch-metal-nightly.rb:1]: The token here should be `retroarch-metal@nightly`

File: Casks/r/retroarch-metal-nightly.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  url "https://buildbot.libretro.com/nightly/apple/osx/universal/RetroArch_Metal.dmg",
+      verified: "buildbot.libretro.com/nightly/"
+  name "RetroArch Metal Nightly"
+  desc "Frontend for emulators, game engines, and media players (Metal graphics API)"
+  homepage "https://www.retroarch.com/"
+
+  depends_on macos: ">= :high_sierra"
+
+  app "RetroArch.app", target: "RetroArch Nightly.app"
```
Comment (bevanjkay) [inline Casks/r/retroarch-metal-nightly.rb:13]: Looking at the `zap` it seems that this doesn't share preferences with the non-nightly version? If that is correct, I agree that it is appropriate to use `target` here to allow both this and the non-nightly version to be installed at the same time.

Comment (bevanjkay) [reply Casks/r/retroarch-metal-nightly.rb:13]: If this is true, can you add a comment explaining that this is the reason for the use of `target`.

Comment (fpscan) [reply Casks/r/retroarch-metal-nightly.rb:13]: It's actually never mention as a RetroArch Nightly in anwywhere, but you can install both in the same system. When I tried on Brew, I thought this is the way have both of them in the same system.

File: Casks/r/retroarch-metal-nightly.rb (hunk 1 above)
Comment (fpscan) [reply Casks/r/retroarch-metal-nightly.rb]: [a71a1b7](https://github.com/Homebrew/homebrew-cask/pull/177874/commits/a71a1b70705de6a283e8303d3432b6adb2409907)

**Review by bevanjkay — COMMENTED** [review]
@fpscan Please also rename the file to `retroarch-metal@nightly.rb`

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**